code/modules/clothing/clothing.dm

```diff
@@ -217,6 +217,23 @@
 		for(var/A in armor)
 			armor[A] -= rand(armor[A]/3, armor[A])
 
+/obj/item/clothing/attack(mob/living/M, mob/living/user, def_zone, var/originator = null)
+	if(iscarbon(user))
+		if(user.a_intent == I_HELP)
+			if(istype(src, /obj/item/clothing/under) || (istype(src, /obj/item/clothing/suit) && !istype(src, /obj/item/clothing/suit/space)))
+				if(ishuman(M) && M.on_fire)
+					var/mob/living/carbon/human/target = M
+					if(isplasmaman(target)) // Cannot put out plasmamen, else they could just go around with a jumpsuit and not need a space suit.
+						visible_message("<span class='warning'>\The [user] attempts to put out the fire on \the [target], but plasmafires are too hot. It is no use.</span>")
+						return
+					else
+						visible_message("<span class='warning'>\The [user] attempts to put out the fire on \the [target] with \the [src].</span>")
+						if(prob(15) || (istype(src, /obj/item/clothing/suit/spaceblanket) && prob(70))) // Blankets are better at putting out fire
```

Comment:
This could be reworked to be more OO: make a `var/extinguish_prob` under the definition of `/obj/item/clothing` with a default value of 15 and override it for the spaceblanket to 70. Then you only need to check `prob(extinguish_prob)`.